cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -466,17 +480,13 @@ var (
 				return nil
 			}
 			var device ttnpb.EndDevice
-			if ttnpb.HasAnyField(ttnpb.TopLevelFields(paths), "root_keys") {
-				device.SupportsJoin = true
-				paths = append(paths, "supports_join")
-			}
 			if err = util.SetFields(&device, setEndDeviceFlags); err != nil {
 				return err
 			}
 			device.Attributes = mergeAttributes(device.Attributes, cmd.Flags())
 			device.EndDeviceIdentifiers = *devID
 
-			isPaths, nsPaths, asPaths, jsPaths := splitEndDeviceSetPaths(device.SupportsJoin, paths...)
+			isPaths, nsPaths, asPaths, jsPaths := splitEndDeviceSetPaths(ttnpb.HasAnyField(paths, setEndDeviceToJS...), paths...)
```

Comment:
I think this actually needs to be SupportsJoin

Reply:
We're here in update. If you pass `--resets_join_nonces=false`, it should go to JS.

Otherwise we require the user to pass `--supports-join` and that's not very user friendly.

Reply:
Then the first argument of splitEndDeviceSetPaths needs to be renamed

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/844/commits/2c8cf0658cb36193194362402098f889a8c5e26c